rocketpy/mathutils/function.py

```diff
@@ -330,22 +338,29 @@ def set_extrapolation(self, method="constant"):
 
     def __set_interpolation_func(self):  # pylint: disable=too-many-statements
         """Defines interpolation function used by the Function. Each
-        interpolation method has its own function with exception of shepard,
-        which has its interpolation/extrapolation function defined in
-        ``Function.__interpolate_shepard__``. The function is stored in
-        the attribute _interpolation_func."""
+        interpolation method has its own function`.
+        The function is stored in the attribute _interpolation_func."""
         interpolation = INTERPOLATION_TYPES[self.__interpolation__]
         if interpolation == 0:  # linear
+            if self.__dom_dim__ == 1:
 
-            def linear_interpolation(
-                x, x_min, x_max, x_data, y_data, coeffs
-            ):  # pylint: disable=unused-argument
-                x_interval = bisect_left(x_data, x)
-                x_left = x_data[x_interval - 1]
-                y_left = y_data[x_interval - 1]
-                dx = float(x_data[x_interval] - x_left)
-                dy = float(y_data[x_interval] - y_left)
-                return (x - x_left) * (dy / dx) + y_left
+                def linear_interpolation(
```

Comment:
I don't really like the idea of defining a function inside of a method, in this case if you define N Functions with linear interpolation, this function will be re-created N times

HOWEVER, this is not a problem being introduced by this PR.

Reply:
I agree with you. In my opinion each interpolation function should be selected by a key value pair instead a if-else or match case, is would be more organized and would have a small (basically negligible since this if-else is small) performance advantage.